workflow/controller/operator.go

```diff
@@ -1679,6 +1679,11 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 			woc.addChildNode(retryNodeName, nodeName)
 			node = nil
 
+			// It has to be one child at least
+			if lastChildNode != nil {
```

Comment:
so `executeTemplate` is the point in the code where we apply "retry tweaks"

Reply:
I found this the best point for it, all the necessary data are here.